pageserver/src/virtual_file/owned_buffers_io/write.rs

```diff
@@ -125,6 +138,29 @@ where
         Ok((chunk_len, FullSlice::must_new(chunk)))
     }
 
+    async fn flush(&mut self, ctx: &RequestContext) -> std::io::Result<()> {
+        let buf = self.buf.take().expect("must not use after an error");
+        let buf_len = buf.pending();
+        if buf_len == 0 {
+            self.buf = Some(buf);
+            return Ok(());
+        }
+        let slice = buf.flush();
+        let (nwritten, slice) = self.writer.write_all(slice, ctx).await?;
+        assert_eq!(nwritten, buf_len);
+        self.buf = Some(Buffer::reuse_after_flush(
+            slice.into_raw_slice().into_inner(),
+        ));
+        Ok(())
+    }
+}
+
+impl<B, Buf, W> BufferedWriter<B, W>
```

Comment:
write_buffered_borrowed now uses the `IoBufAligned` marker trait to guarantee the alignment.